source/extensions/common/aws/utility.cc

```diff
@@ -222,6 +222,22 @@ Utility::joinCanonicalHeaderNames(const std::map<std::string, std::string>& cano
   });
 }
 
+std::string Utility::getSTSEndpoint(absl::string_view region) {
+  if (region == "cn-northwest-1" || region == "cn-north-1") {
+    return fmt::format("sts.{}.amazonaws.com.cn", region);
+  }
+#ifdef ENVOY_SSL_FIPS
```

Comment:
I wonder if this should ultimately be configurable vs. hard coded at compile time? Seems more flexible to me.

Reply:
My thinking is that it is ok to hard-code at compile time as the default option, whether the binary was built with BoringSSL or BoringSSL-FIPS.

Configuration option as enhancement would also be useful for testing out or flexibility. But as of today there is not much configurations sent down from extensions (AWS Lambda or AWS Request Signing) towards the aws credentials signing utility other than just the region. Also if a common control plane is used for FIPS or non-FIPS Envoy clients, it will be additional data that client Envoy needs to tell control plane in the node information that it can support FIPS mode.

So I though natively Envoy binary making this decision seemed simpler.

--

In saying that I don't see #ifdef `ENVOY_SSL_FIPS` being used widely in Envoy.

Reply:
OK it's fine with me if you want to start with this and you can see how it goes.

Reply:
Thanks Matt for approval and review.
Meanwhile, following the news about Bitdrift. Looking forward for more updates.